### ISSUE TITLE
Can't compile LTS refspdk on fedora32

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#1717]
```
SPDK: LTS (v20.01-54-gb2808069e)
OS: Fedora32 (gcc: 10.2.1, kernel: 5.9.11-100.fc32.x86_64)
```

While compiling against the LTS release, `install_refspdk()`, called via `vm_setup.sh`, fails with the following:

```
/root/spdk_abi_latest/dpdk/drivers/vdpa/ifc/base/ifcvf.h:16: error: "VIRTIO_F_IOMMU_PLATFORM" redefined [-Werror]
   16 | #define VIRTIO_F_IOMMU_PLATFORM  33
      |
In file included from /usr/include/linux/virtio_net.h:30,
                 from /root/spdk_abi_latest/dpdk/drivers/vdpa/ifc/ifcvf_vdpa.c:11:
/usr/include/linux/virtio_config.h:78: note: this is the location of the previous definition
   78 | #define VIRTIO_F_IOMMU_PLATFORM  VIRTIO_F_ACCESS_PLATFORM
      |
  PMDINFO otx_cryptodev.o.pmd.c
  CC otx_cryptodev.o.pmd.o
  LD otx_cryptodev.o
cc1: all warnings being treated as errors
  CC otx_cryptodev_capabilities.o
make[7]: *** [/root/spdk_abi_latest/dpdk/mk/internal/rte.compile-pre.mk:116: ifcvf_vdpa.o] Error 1
make[6]: *** [/root/spdk_abi_latest/dpdk/mk/rte.subdir.mk:37: ifc] Error 2
make[5]: *** [/root/spdk_abi_latest/dpdk/mk/rte.subdir.mk:37: vdpa] Error 2
make[5]: *** Waiting for unfinished jobs....
```
Without the `werror` enabled:
```
 LIB libspdk_bdev_null.a
  CC module/bdev/split/vbdev_split.o
  LIB libspdk_bdev_gpt.a
  CC module/bdev/zone_block/vbdev_zone_block.o
  LIB libspdk_bdev_delay.a
  CC module/bdev/crypto/vbdev_crypto.o
  CC module/bdev/ocf/ctx.o
  CC module/bdev/compress/vbdev_compress.o
  LIB libspdk_bdev_rpc.a
  CC module/bdev/ocf/data.o
  CC module/bdev/split/vbdev_split_rpc.o
  CC module/bdev/ocf/stats.o
  LIB libspdk_bdev_lvol.a
  CC module/bdev/ocf/utils.o
stats.c: In function ‘vbdev_ocf_stats_get’:
stats.c:43:11: warning: implicit declaration of function ‘ocf_core_get’; did you mean ‘ocf_ctx_get’? [-Wimplicit-function-declaration]
   43 |  status = ocf_core_get(cache, core_id, &core);
      |           ^~~~~~~~~~~~
      |           ocf_ctx_get
stats.c: In function ‘vbdev_ocf_stats_write_json’:
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:65:2: note: in expansion of macro ‘WJSON_STAT’
   65 |  WJSON_STAT(w, stats, usage, occupancy, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:65:2: note: in expansion of macro ‘WJSON_STAT’
   65 |  WJSON_STAT(w, stats, usage, occupancy, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:66:2: note: in expansion of macro ‘WJSON_STAT’
   66 |  WJSON_STAT(w, stats, usage, free, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:66:2: note: in expansion of macro ‘WJSON_STAT’
   66 |  WJSON_STAT(w, stats, usage, free, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:67:2: note: in expansion of macro ‘WJSON_STAT’
   67 |  WJSON_STAT(w, stats, usage, clean, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:67:2: note: in expansion of macro ‘WJSON_STAT’
   67 |  WJSON_STAT(w, stats, usage, clean, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:68:2: note: in expansion of macro ‘WJSON_STAT’
   68 |  WJSON_STAT(w, stats, usage, dirty, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:68:2: note: in expansion of macro ‘WJSON_STAT’
   68 |  WJSON_STAT(w, stats, usage, dirty, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:72:2: note: in expansion of macro ‘WJSON_STAT’
   72 |  WJSON_STAT(w, stats, reqs, rd_hits, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:72:2: note: in expansion of macro ‘WJSON_STAT’
   72 |  WJSON_STAT(w, stats, reqs, rd_hits, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:73:2: note: in expansion of macro ‘WJSON_STAT’
   73 |  WJSON_STAT(w, stats, reqs, rd_partial_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:73:2: note: in expansion of macro ‘WJSON_STAT’
   73 |  WJSON_STAT(w, stats, reqs, rd_partial_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:74:2: note: in expansion of macro ‘WJSON_STAT’
   74 |  WJSON_STAT(w, stats, reqs, rd_full_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:74:2: note: in expansion of macro ‘WJSON_STAT’
   74 |  WJSON_STAT(w, stats, reqs, rd_full_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:75:2: note: in expansion of macro ‘WJSON_STAT’
   75 |  WJSON_STAT(w, stats, reqs, rd_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:75:2: note: in expansion of macro ‘WJSON_STAT’
   75 |  WJSON_STAT(w, stats, reqs, rd_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:76:2: note: in expansion of macro ‘WJSON_STAT’
   76 |  WJSON_STAT(w, stats, reqs, wr_hits, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:76:2: note: in expansion of macro ‘WJSON_STAT’
   76 |  WJSON_STAT(w, stats, reqs, wr_hits, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:77:2: note: in expansion of macro ‘WJSON_STAT’
   77 |  WJSON_STAT(w, stats, reqs, wr_partial_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:77:2: note: in expansion of macro ‘WJSON_STAT’
   77 |  WJSON_STAT(w, stats, reqs, wr_partial_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:78:2: note: in expansion of macro ‘WJSON_STAT’
   78 |  WJSON_STAT(w, stats, reqs, wr_full_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:78:2: note: in expansion of macro ‘WJSON_STAT’
   78 |  WJSON_STAT(w, stats, reqs, wr_full_misses, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:79:2: note: in expansion of macro ‘WJSON_STAT’
   79 |  WJSON_STAT(w, stats, reqs, wr_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:79:2: note: in expansion of macro ‘WJSON_STAT’
   79 |  WJSON_STAT(w, stats, reqs, wr_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:80:2: note: in expansion of macro ‘WJSON_STAT’
   80 |  WJSON_STAT(w, stats, reqs, rd_pt, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:80:2: note: in expansion of macro ‘WJSON_STAT’
   80 |  WJSON_STAT(w, stats, reqs, rd_pt, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:81:2: note: in expansion of macro ‘WJSON_STAT’
   81 |  WJSON_STAT(w, stats, reqs, wr_pt, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:81:2: note: in expansion of macro ‘WJSON_STAT’
   81 |  WJSON_STAT(w, stats, reqs, wr_pt, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:82:2: note: in expansion of macro ‘WJSON_STAT’
   82 |  WJSON_STAT(w, stats, reqs, serviced, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:82:2: note: in expansion of macro ‘WJSON_STAT’
   82 |  WJSON_STAT(w, stats, reqs, serviced, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:83:2: note: in expansion of macro ‘WJSON_STAT’
   83 |  WJSON_STAT(w, stats, reqs, total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:83:2: note: in expansion of macro ‘WJSON_STAT’
   83 |  WJSON_STAT(w, stats, reqs, total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:87:2: note: in expansion of macro ‘WJSON_STAT’
   87 |  WJSON_STAT(w, stats, blocks, core_volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:87:2: note: in expansion of macro ‘WJSON_STAT’
   87 |  WJSON_STAT(w, stats, blocks, core_volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:88:2: note: in expansion of macro ‘WJSON_STAT’
   88 |  WJSON_STAT(w, stats, blocks, core_volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:88:2: note: in expansion of macro ‘WJSON_STAT’
   88 |  WJSON_STAT(w, stats, blocks, core_volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:89:2: note: in expansion of macro ‘WJSON_STAT’
   89 |  WJSON_STAT(w, stats, blocks, core_volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:89:2: note: in expansion of macro ‘WJSON_STAT’
   89 |  WJSON_STAT(w, stats, blocks, core_volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:90:2: note: in expansion of macro ‘WJSON_STAT’
   90 |  WJSON_STAT(w, stats, blocks, cache_volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:90:2: note: in expansion of macro ‘WJSON_STAT’
   90 |  WJSON_STAT(w, stats, blocks, cache_volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:91:2: note: in expansion of macro ‘WJSON_STAT’
   91 |  WJSON_STAT(w, stats, blocks, cache_volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:91:2: note: in expansion of macro ‘WJSON_STAT’
   91 |  WJSON_STAT(w, stats, blocks, cache_volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:92:2: note: in expansion of macro ‘WJSON_STAT’
   92 |  WJSON_STAT(w, stats, blocks, cache_volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:92:2: note: in expansion of macro ‘WJSON_STAT’
   92 |  WJSON_STAT(w, stats, blocks, cache_volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:93:2: note: in expansion of macro ‘WJSON_STAT’
   93 |  WJSON_STAT(w, stats, blocks, volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:93:2: note: in expansion of macro ‘WJSON_STAT’
   93 |  WJSON_STAT(w, stats, blocks, volume_rd, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:94:2: note: in expansion of macro ‘WJSON_STAT’
   94 |  WJSON_STAT(w, stats, blocks, volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:94:2: note: in expansion of macro ‘WJSON_STAT’
   94 |  WJSON_STAT(w, stats, blocks, volume_wr, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:95:2: note: in expansion of macro ‘WJSON_STAT’
   95 |  WJSON_STAT(w, stats, blocks, volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:95:2: note: in expansion of macro ‘WJSON_STAT’
   95 |  WJSON_STAT(w, stats, blocks, volume_total, "4KiB blocks");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:99:2: note: in expansion of macro ‘WJSON_STAT’
   99 |  WJSON_STAT(w, stats, errors, core_volume_rd, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:99:2: note: in expansion of macro ‘WJSON_STAT’
   99 |  WJSON_STAT(w, stats, errors, core_volume_rd, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:100:2: note: in expansion of macro ‘WJSON_STAT’
  100 |  WJSON_STAT(w, stats, errors, core_volume_wr, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:100:2: note: in expansion of macro ‘WJSON_STAT’
  100 |  WJSON_STAT(w, stats, errors, core_volume_wr, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:101:2: note: in expansion of macro ‘WJSON_STAT’
  101 |  WJSON_STAT(w, stats, errors, core_volume_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:101:2: note: in expansion of macro ‘WJSON_STAT’
  101 |  WJSON_STAT(w, stats, errors, core_volume_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:102:2: note: in expansion of macro ‘WJSON_STAT’
  102 |  WJSON_STAT(w, stats, errors, cache_volume_rd, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:102:2: note: in expansion of macro ‘WJSON_STAT’
  102 |  WJSON_STAT(w, stats, errors, cache_volume_rd, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:103:2: note: in expansion of macro ‘WJSON_STAT’
  103 |  WJSON_STAT(w, stats, errors, cache_volume_wr, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:103:2: note: in expansion of macro ‘WJSON_STAT’
  103 |  WJSON_STAT(w, stats, errors, cache_volume_wr, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:104:2: note: in expansion of macro ‘WJSON_STAT’
  104 |  WJSON_STAT(w, stats, errors, cache_volume_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:104:2: note: in expansion of macro ‘WJSON_STAT’
  104 |  WJSON_STAT(w, stats, errors, cache_volume_total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:21: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                     ^
stats.c:105:2: note: in expansion of macro ‘WJSON_STAT’
  105 |  WJSON_STAT(w, stats, errors, total, "Requests");
      |  ^~~~~~~~~~
stats.c:55:54: error: ‘struct ocf_stat’ has no member named ‘percent’
   55 |   stats->group.field.percent / 10, stats->group.field.percent % 10); \
      |                                                      ^
stats.c:105:2: note: in expansion of macro ‘WJSON_STAT’
  105 |  WJSON_STAT(w, stats, errors, total, "Requests");
      |  ^~~~~~~~~~
make[3]: *** [/root/spdk_abi_latest/mk/spdk.common.mk:374: stats.o] Error 1
make[3]: *** Waiting for unfinished jobs....
  CC module/bdev/compress/vbdev_compress_rpc.o
  LIB libspdk_bdev_split.a
  CC module/bdev/uring/bdev_uring.o
  CC module/bdev/uring/bdev_uring_rpc.o
  CC module/bdev/zone_block/vbdev_zone_block_rpc.o
  CC module/bdev/aio/bdev_aio.o
make[2]: *** [/root/spdk_abi_latest/mk/spdk.subdirs.mk:44: ocf] Error 2
make[2]: *** Waiting for unfinished jobs....
  CC module/bdev/aio/bdev_aio_rpc.o
  CC module/bdev/crypto/vbdev_crypto_rpc.o
  CC module/bdev/nvme/nvme_rpc.o
  CC module/bdev/raid/bdev_raid_rpc.o
  CC module/bdev/raid/raid0.o
  LIB libspdk_bdev_zone_block.a
  CC module/bdev/nvme/common.o
  CC module/bdev/nvme/bdev_ocssd.o
  CC module/bdev/nvme/bdev_ocssd_rpc.o
  LIB libspdk_bdev_uring.a
  CC module/bdev/nvme/vbdev_opal.o
  CC module/bdev/nvme/vbdev_opal_rpc.o
  LIB libspdk_bdev_aio.a
  LIB libspdk_bdev_raid.a
  CC module/bdev/nvme/bdev_nvme_cuse_rpc.o
  LIB libspdk_bdev_nvme.a
  LIB libspdk_bdev_compress.a
  LIB libspdk_bdev_crypto.a
make[1]: *** [/root/spdk_abi_latest/mk/spdk.subdirs.mk:44: bdev] Error 2
make: *** [/root/spdk_abi_latest/mk/spdk.subdirs.mk:44: module] Error 2
```
Odly enough, after `make clean`, compilation fails at a different place:
```
src/ocf/concurrency/ocf_metadata_concurrency.c: In function ‘ocf_metadata_concurrency_init’:
src/ocf/concurrency/ocf_metadata_concurrency.c:14:6: error: void value not ignored as it ought to be
   14 |  err = env_spinlock_init(&metadata_lock->eviction);
      |      ^
src/ocf/concurrency/ocf_metadata_concurrency.c:24:7: error: void value not ignored as it ought to be
   24 |   err = env_spinlock_init(&metadata_lock->partition[i]);
      |       ^
src/ocf/concurrency/ocf_cache_line_concurrency.c: In function ‘ocf_cache_line_concurrency_init’:
src/ocf/concurrency/ocf_cache_line_concurrency.c:121:9: error: void value not ignored as it ought to be
  121 |   error = env_spinlock_init(&c->waiters_lsts[i].lock);
      |         ^
cc -g  -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wmissing-declarations -fno-strict-aliasing -I/root/spdk_abi_latest/include -march=native -Wformat -Wformat-security -D_GNU_SOURCE -fPIC -fstack-protector -fno-common -I/root/spdk_abi_latest/isa-l/.. -DDEBUG -O0 -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined -DSPDK_GIT_COMMIT=b2808069e -pthread -Wno-pointer-sign -Wstrict-prototypes -Wold-style-definition -std=gnu99 -I/root/spdk_abi_latest/dpdk/build/include -Wno-deprecated-declarations -I/root/spdk_abi_latest/lib/env_ocf -I/root/spdk_abi_latest/lib/env_ocf/include -w   -c -o src/ocf/ocf_freelist.o src/ocf/ocf_freelist.c
make[3]: *** [<builtin>: src/ocf/concurrency/ocf_metadata_concurrency.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [<builtin>: src/ocf/concurrency/ocf_cache_line_concurrency.o] Error 1
src/ocf/ocf_freelist.c: In function ‘ocf_freelist_init’:
src/ocf/ocf_freelist.c:392:7: error: void value not ignored as it ought to be
  392 |   if (env_spinlock_init(&freelist->lock[i]))
      |       ^~~~~~~~~~~~~~~~~
make[3]: *** [<builtin>: src/ocf/ocf_freelist.o] Error 1
  LIB libspdk_vmd.a
  LIB libspdk_util.a
make[2]: *** [Makefile:76: all] Error 2
make[1]: *** [/root/spdk_abi_latest/mk/spdk.subdirs.mk:44: env_ocf] Error 2
make[1]: *** Waiting for unfinished jobs....
```
./configure is run with the following arguments:
```
--enable-debug
--enable-werror
--with-rdma
--with-fio=/usr/src/fio
--with-iscsi-initiator
--with-nvme-cuse
--with-pmdk
--with-reduce
--with-rbd
--with-crypto
--with-ocf
--enable-ubsan
--enable-asan
--with-fuse
--with-uring
```

**Comment by tomzawadzki** [comment]
Hi @mikeBashStuff, I'm running F32 too.
```
OS: 5.8.7-200.fc32.x86_64
GCC:  gcc version 10.2.1 20200723 (Red Hat 10.2.1-1) (GCC)
```

On SPDK LTS using `./configure --enable-debug --enable-werror --with-rdma --with-fio=/usr/src/fio --with-iscsi-initiator --with-nvme-cuse --with-pmdk --with-reduce --with-rbd --with-crypto --with-ocf --enable-ubsan --enable-asan --with-fuse --with-uring`

Produces issues mentioned in #1493:
- dpdk build, can be fixed by applying 970c6d099e3db9ca67b19d8cdb56273da6818735 and 1415e38411fea0499fc90d6d1af7be75aff0e668
- `nvme_pcie_ut.c` fixed by 7641283387e9ed237bbb0824f2a114ad65684172 or worked around using `./configure --disable-tests`. Later should not matter for ABI testing.
After applying the above, compilation passes.

The above basically highlight lack of compatibility with gcc 10 on LTS, which came later this year. @jimharris should the three patches above find their way into `v20.01.x` branch ?

Either way, I've not found any issues with OCF. @mikeBashStuff have you made sure to clean up the repo and update submodules ?

**Comment by mikeBashStuff** [comment]
@tomzawadzki Ah, right, I always forget that checkout doesn't touch submodules by default (need to finally alias that `--recurse-submodules=yes` :D). Anyway, with submodules checked out at proper commits I can confirm that the #1493 is seen instead (this is also what  is seen in the vagrant-build-images job for fedora32).

Let's keep this issue opened until it's decided if any backports are needed. We can close it afterwards. :)

**Comment by tomzawadzki** [comment]
[Bug scrub] Yes, we should backport the 3 patches to the v20.01.x branch, without doing the next maintenance LTS release.

**Comment by tomzawadzki** [comment]
Patches were merged to v20.01.x branch
